smdebug/tensorflow/keras.py

```diff
@@ -497,7 +516,8 @@ def _on_any_batch_begin(self, batch, mode, logs=None):
                 # if saving metric, writer may not be initialized as a result
                 self._initialize_writers()
 
-            self._add_callbacks(mode)
+            if not is_tf_version_2x() or (is_tf_version_2x() and not tf.executing_eagerly()):
```

Comment:
else?

Reply:
else, we dont add callbacks because TF 2.x eager mode doesn't support fetches. else case is TF 2.x and eager mode